### DISCUSSION TITLE
Is it possible to add SBC Dual Channel HD (SBC XQ) 551 kbps option to --sbc-quality=?

### DISCUSSION THREAD

**Discussion by DaPiMan** [Arkq/bluez-alsa#548] [Q&A] [answered]
Is it possible to add another option flag for option "--sbc-quality=NUM" for SBC Dual Channel HD (SBC XQ) 551 kbps?

For testing I modified bitpool=38 to bitpool=47 in codec-sbc.c (code chunk below) and recompiled successfully. My raspberry pi4 got a SBC XQ connection to by bluetooth reciever successfully. I however don't really have the knowledge on creating another flag option for the compiled app as I would like to be able to switch between 452kbps and 551kbps. I got the idea from here: https://lineageos.org/engineering/Bluetooth-SBC-XQ/

        if (quality == SBC_QUALITY_XQ) {
                /* Check whether XQ is possible. If not, downgrade to high quality. */
                if (conf->frequency == SBC_SAMPLING_FREQ_44100 &&
                                conf->channel_mode == SBC_CHANNEL_MODE_DUAL_CHANNEL &&
                                conf->block_length == SBC_BLOCK_LENGTH_16 &&
                                conf->subbands == SBC_SUBBANDS_8 &&
                                conf->allocation_method == SBC_ALLOCATION_LOUDNESS)
                        **bitpool = 38;**
                else {
                        warn("Unable to use SBC XQ, downgrading to high quality");
                        quality = SBC_QUALITY_HIGH;
                }

**Comment by arkq** [comment, marked as answer]
Since commit 6950e8b there is an extra mode for SBC quality option (SBC XQ+):

```sh
bluealsa --sbc-quality=xq+
```

Please, check this PR: https://github.com/Arkq/bluez-alsa/pull/549

**Reply by DaPiMan** [reply]
Perfect!

sudo service bluezalsa status
● bluezalsa.service - BluezAlsa proxy
   Loaded: loaded (/lib/systemd/system/bluezalsa.service; enabled; vendor preset: enabled)
   Active: active (running) since Sat 2022-03-05 04:13:33 UTC; 3s ago
 Main PID: 6608 (bluealsa)
    Tasks: 3 (limit: 2128)
   CGroup: /system.slice/bluezalsa.service
           └─6608 /usr/bin/bluealsa -p a2dp-source -p a2dp-sink -c sbc -c aptx -c aptx-hd --sbc-quality=xq+ --a2dp-force-audio-cd

Mar 05 04:13:33 audiostreamer systemd[1]: Started BluezAlsa proxy.
Mar 05 04:13:33 audiostreamer bluealsa[6608]: /usr/bin/bluealsa: **I: Activating SBC Dual Channel HD (SBC XQ+)**
